### PR TITLE
Fix NaN in ROD angle calculation for parallel vectors

### PR DESCRIPTION
**Pull request by Himerometra**
## Summary

Fix pyod.models.rod.angle returning NaN for valid parallel or antiparallel vectors due to floating-point roundoff, overflow, or underflow.

The vectors are scaled by their largest absolute component before computing the cosine. The final cosine is clipped to [-1, 1] before calling arccos. Zero vectors retain the existing NaN result.

## Testing

Added regression coverage for:

- roundoff beyond both cosine boundaries;
- ordinary parallel, orthogonal, and antiparallel vectors;
- extreme float32 and float64 magnitudes;
- zero-vector behavior.

Test command:

    python -m pytest pyod/test/test_rod_angle.py pyod/test/test_rod.py -q
    38 passed, 7 warnings

No existing issue; this was found through direct numerical edge-case testing.

### CODE REVIEW

File: pyod/models/rod.py [hunk 1]
```diff
@@ -52,8 +52,16 @@ def angle(v1, v2):
     -------
     angle : float, the angle
     """
-    return np.arccos(np.dot(v1, v2) /
-                     (np.linalg.norm(v1) * np.linalg.norm(v2)))
+    v1 = np.asarray(v1)
+    v2 = np.asarray(v2)
+    v1_scale = np.max(np.abs(v1))
```
[the remark this answers is not in the document]
Comment (Himerometra) [reply pyod/models/rod.py:60]: Fixed in `da3463b`. The inputs are now promoted to a common floating dtype before taking absolute maxima, preventing overflow for signed integer minimum values. I also added parallel and antiparallel regression coverage using `np.int8` minimum; the related suite passes with 40 tests.
